spec/spec.md

```diff
@@ -39,17 +39,27 @@ Presentation Exchange is a draft specification being developed within the Decent
 Term | Definition
 :--- | :---------
 Decentralized Identifier (DID) | Unique ID string and PKI metadata document format for describing the cryptographic keys and other fundamental PKI values linked to a unique, user-controlled, self-sovereign identifier in a target system (i.e. blockchain, distributed ledger).
-Subject | The entity that submits proofs to a Verifier to satisfy the requirements described in a Presentation Definition
-Verifier | The entity that defines what proofs they require from a Subject (via a Presentation Definition) in order to proceed with an interaction.
+Prover | The entity that submits proofs to a Verifier to satisfy the requirements described in a Presentation Definition
+Verifier | The entity that defines what proofs they require from a Prover (via a Presentation Definition) in order to proceed with an interaction.
 
 ## Presentation Definition
 
-Presentation Definitions are objects generate to articulate what proofs an entity requires to make a decision about an interaction with a Subject. Presentation Definitions are composed of inputs, which describe the forms and details of the proofs they require, and and optional set of selection rules, to allow Subjects flexibility in cases where many different types of proofs may satisfy an input requirement.
+Presentation Definitions articulate what proofs Verifier requires. Often, but not always, they help the Verifier decide how or whether to interact with a Prover. Presentation Definitions are composed of inputs, which describe the forms and details of the proofs they require, and and optional set of selection rules, to allow Subjects flexibility in cases where many different types of proofs may satisfy an input requirement.
 
 ::: example Presentation Definition - all features exercised
-```json
+```jsonc
 
 {
+    // TODO: The presentation definition MUST include a nonce. If it doesn't,
```

Comment:
this is handled in the VP by the use of the `challenge` and `domain` values... if they are not obviously included here, we need to do that.

Reply:
Not every embed target is a VP, so we may need to include a nonce. I will try to add this as a MAY, given you may want to publish these statically for some reasons.

Reply:
the nonce does not go in the embed target... it goes in the https://identity.foundation/presentation-exchange/#presentation-submission

Which is a compliant "Verifiable Presentation". see: https://w3c.github.io/vc-data-model/#example-2-a-simple-example-of-a-verifiable-presentation

Reply:
Okay. As a data format question, that's fine. And I get that this is a format spec, not a protocol spec. But the value that we end up signing MUST or STRONGLY SHOULD be supplied by the verifier. Filling out the presentation submission with this value but having the value be invented by the prover accomplishes zero, cybersecurity-wise. (It's conceivable that the value could derive from the latest state hash of a continually evolving blockchain or similar, but unless the verifier can prove it didn't exist before the prover created the proof, it's inadequate.)

Reply:
exactly right!, the prover must be proving over a `challenge` (@csuwildcat please use the vc-data-model terminology)... and the `challenge` must be supplied by the verifier, in order to prevent replay attacks.